finality-aleph/src/party/member.rs

```diff
@@ -44,3 +59,23 @@ pub fn task<
     let handle = spawn_handle.spawn_essential("aleph/consensus_session_member", task);
     Task::new(handle, stop)
 }
+
+fn rotate_saved_unit_files(stash_path: &Path, session_id: u32) -> Result<(File, File), io::Error> {
+    let extension = ".abfst";
+    let session_path = stash_path.join(format!("{}", session_id));
+    fs::create_dir_all(&session_path)?;
+    let index = fs::read_dir(&session_path)
+        .unwrap()
+        .filter_map(|r| r.ok())
+        .filter_map(|x| x.file_name().into_string().ok())
+        .filter_map(|s| u64::from_str(s.strip_suffix(extension)?).ok())
+        .max();
+    let load_path = match index {
+        Some(index) => session_path.join(format!("{}{}", index, extension)),
+        None => "/dev/null".into(),
+    };
+    let load_file = File::open(load_path)?;
```

Comment:
There is also one more thing. AlephBFT needs states from all runs in current session. Saver saves only newly created units to a file. So just loading the last file is not enough. We need to load all files from this session and also in right order. So sorted by the number of run.